homeassistant/components/lcn/climate.py

```diff
@@ -0,0 +1,139 @@
+
+    async_add_entities(devices)
+
+
+class LcnClimate(LcnDevice, ClimateDevice):
+    """Representation of a LCN climate device."""
+
+    def __init__(self, config, address_connection):
+        """Initialize of a LCN climate device."""
+        super().__init__(config, address_connection)
+
+        self.variable = self.pypck.lcn_defs.Var[config[CONF_SOURCE]]
+        self.setpoint = self.pypck.lcn_defs.Var[config[CONF_SETPOINT]]
+        self.unit = self.pypck.lcn_defs.VarUnit.parse(
+            config[CONF_UNIT_OF_MEASUREMENT])
+
+        self.regulator_id = \
+            self.pypck.lcn_defs.Var.to_set_point_id(self.setpoint)
+        self.is_lockable = config[CONF_LOCKABLE]
+        self._max_temp = config[CONF_MAX_TEMP]
+        self._min_temp = config[CONF_MIN_TEMP]
+
+        self._current_temperature = None
+        self._target_temperature = None
+        self._is_on = True
+
+    async def async_added_to_hass(self):
+        """Run when entity about to be added to hass."""
+        await super().async_added_to_hass()
+        self.hass.async_create_task(
+            self.address_connection.activate_status_request_handler(
+                self.variable))
+        self.hass.async_create_task(
+            self.address_connection.activate_status_request_handler(
+                self.setpoint))
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        support = const.SUPPORT_TARGET_TEMPERATURE
+        if self.is_lockable:
+            support |= const.SUPPORT_ON_OFF
+        return support
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self.unit.value
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def is_on(self):
+        """Return true if the device is on."""
+        return self._is_on
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._max_temp
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._min_temp
+
+    async def async_turn_on(self):
+        """Turn on."""
+        self._is_on = True
+        self.address_connection.lock_regulator(self.regulator_id, False)
+        await self.async_update_ha_state()
+
+    async def async_turn_off(self):
+        """Turn off."""
+        self._is_on = False
+        self.address_connection.lock_regulator(self.regulator_id, True)
+        self._target_temperature = None
+        await self.async_update_ha_state()
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
```

Comment:
Store the temperature in a local variable first. Then check if it's None, and if so return.